### PR TITLE
Add dispatchEvent method and CreateEvent function

### PR DESCRIPTION
**Pull request by arvitaly**
Added dispatchEvent for window and BasicNode.
Added CreateEvent function for custom event

### CODE REVIEW

File: events.go [hunk 1]
```diff
@@ -137,6 +137,22 @@ type Event interface {
 // concrete event types.
 type BasicEvent struct{ *js.Object }
 
+type EventInit struct {
```
Comment (shurcooL) [inline events.go]: Perhaps it'd be better (more idiomatic) to name this struct `EventOptions`? Similar to [`PositionOptions`](https://github.com/arvitaly/go-js-dom/blob/9af2ce4cd4e5d5782e90b05dca66f73c88f97582/dom.go#L1242) or [`imports.Options`](https://godoc.org/golang.org/x/tools/imports#Options).

**Comment by arvitaly** [comment]
I took EventInit from Mozilla docs:
https://developer.mozilla.org/en-US/docs/Web/API/Event/Event
But, most probably, you're right.

File: events.go [hunk 2]
```diff
@@ -137,6 +137,22 @@ type Event interface {
...
 type BasicEvent struct{ *js.Object }
 
+type EventOptions struct {
+	Bubbles    bool
+	Cancelable bool
+}
+
+func CreateEvent(typeArg string, opts *EventOptions) *BasicEvent {
+	if opts == nil {
+		opts = &EventOptions{Bubbles: false, Cancelable: false}
```
Comment (dominikh) [inline events.go]: `&EventOptions{}` suffices

File: events.go [hunk 3]
```diff
@@ -137,6 +137,22 @@ type Event interface {
 // concrete event types.
 type BasicEvent struct{ *js.Object }
 
+type EventOptions struct {
+	Bubbles    bool
+	Cancelable bool
+}
+
+func CreateEvent(typeArg string, opts *EventOptions) *BasicEvent {
```
Comment (dominikh) [inline events.go]: Do we want *EventOptions or EventOptions? *EventOptions saves some characters when using the defaults, but EventOptions means we don't need the nil check. Geolocation uses PositionOptions, but there isn't an implementation of Geolocation yet (in this package, anyway) so we could possibly change it. The two should be the same, though.

@shurcooL, what's your opinion?

Comment (dominikh) [inline events.go]: Rename typeArg to typ

Comment (shurcooL) [inline events.go]: I'd prefer `EventOptions`, it's simpler and makes adding/removing options a smaller diff.

**Comment by shurcooL** [comment]
What's the status of this PR? @dominikh, it looks like all of our feedback has been addressed, is that so?

**Comment by dominikh** [comment]
Merged as 2a9f3d0.

**Pull request closed without merging**